File: ABTO_iPhone_newVoipSDK_trial_x64(20160923)/Sample_1/Sample_1/UI/ImMessageViewController.m
//
//  ImViewController.m
//  Sample_1
//
//  Created user on 8/13/12.
//  Copyright (c) 2012 Abto Software. All rights reserved.
//

#import "ImMessageViewController.h"

@implementation ImMessageViewController

- (id)initWithContact:(NSString *)contact andText:(NSString *)text {
    self = [super init];
    if (self) {
        imContact = contact;
        imText = text;
        isNew = NO;
    }
    
    return self;
}

- (id)init {
    self = [super init];
    if (self) {
        imContact = nil;
        imText = nil;
        isNew = YES;
    }
    
    return self;
}

- (void)didReceiveMemoryWarning {
    // Releases the view if it doesn't have a superview.
    [super didReceiveMemoryWarning];
    
    // Release any cached data, images, etc that aren't in use.
}

#pragma mark - View lifecycle

- (void)viewDidLoad {
    [super viewDidLoad];
    
    UIBarButtonItem *barRightNavigationItem = [[UIBarButtonItem alloc] initWithTitle:@"Send" style:UIBarButtonItemStyleBordered target:self action:@selector(onButtonSend:)];
    
    [self.navigationController.navigationBar setTranslucent:YES];
    self.navigationItem.rightBarButtonItem = barRightNavigationItem;
    self.navigationItem.leftBarButtonItem.action = @selector(onButtonBack:);
    
    buttonDone = self.navigationItem.rightBarButtonItem;
    // Do any additional setup after loading the view from its nib.
}

- (BOOL)shouldAutorotateToInterfaceOrientation:(UIInterfaceOrientation)interfaceOrientation {
    // Return YES for supported orientations
    return (interfaceOrientation == UIInterfaceOrientationPortrait);
}

- (BOOL)textViewShouldEndEditing:(UITextView *)textView {
    [self->textView resignFirstResponder];
    return YES;
}

-(void)textViewDidBeginEditing:(UITextView *)textView {
    buttonDone.enabled = YES;
    buttonDone.title = @"Send";
    if (!isNew) {
        self->textView.text = @"";
    }
    isNew = YES;
    [self->textView setFrame:CGRectMake(20, 102, 280, 130)];
}

- (void)textViewDidEndEditing:(UITextView *)textView {
    [self->textView setFrame:CGRectMake(20, 102, 280, 338)];
}

- (void)textFieldDidBeginEditing:(UITextField *)textField {
    buttonDone.enabled = YES;
    buttonDone.title = @"Send";
    if (!isNew) {
        self->textView.text = @"";
    }
    isNew = YES;
}

- (void)textFieldDidEndEditing:(UITextField *)textField {
}

- (BOOL)textFieldShouldReturn:(UITextField *)textField {
    [textField resignFirstResponder];
    return YES;
}

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    
    if (imContact && imText) {
        [textView setText:imText];
        [contactAddressBar setText:imContact];
        buttonDone.title = @"Reply";
    } else {
        buttonDone.enabled = NO;
    }
}

- (IBAction)onButtonBack:(id)sender {
    if ([contactAddressBar isFirstResponder]) {
        [contactAddressBar resignFirstResponder];
    }
    if ([textView isFirstResponder]) {
        [textView resignFirstResponder];
    }

    [self.navigationController popViewControllerAnimated:YES];
}

- (IBAction)onButtonSend:(id)sender {
    if (isNew) {
        if ([contactAddressBar.text isEqualToString:@""]) {
            [contactAddressBar becomeFirstResponder];
            return;
        }
        
        if ([textView.text isEqualToString:@""]) {
            [textView becomeFirstResponder];
            return;
        }

        NSString *to = contactAddressBar.text;
        NSString *text = textView.text;
        
        abtoAppDelegate* appDelegate = [abtoAppDelegate sharedInstance];
        AbtoPhoneInterface* phone = appDelegate.phone;
        
        if ([phone sendTextMessage:to withBody:text]) {
            ImMessage *message = [ImMessage new];
            
            message->text = text;
            message->from = phone.config.regUser;
            message->to = to;
            message->isRead = YES;
            
            [appDelegate addMessage:message];
            [appDelegate storeIms];
        }

        [self.navigationController popViewControllerAnimated:YES];
    } else {
        isNew = YES;
        buttonDone.title = @"Send";
        textView.text = @"";
    }
}

- (void)dealloc {
    imContact = nil;
    imText = nil;
}


@end

File: ABTO_iPhone_newVoipSDK_trial_x64(20160923)/SampleSwift/SampleSwift/ImMessageViewController.swift
//
//  ImMessageViewController.swift
//  SampleSwift
//
//  Created by Nazariy Vlizlo on 10/26/15.
//  Copyright © 2015 ABTO Software. All rights reserved.
//

import UIKit

class ImMessageViewController: UIViewController, UITextViewDelegate, UITextFieldDelegate {
    
    @IBOutlet weak var contactAdressBar: UITextField!
    @IBOutlet weak var textView: UITextView!
    
    var isReadyToSent = false
    var isNew = false
    
    var imContact: String?
    var imText: String?
    
    init(contact: String, text: String) {
        imContact = contact
        imText = text
        isNew = false
        super.init(nibName: nil, bundle: nil)
    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
    }
    
    init() {
        super.init(nibName: nil, bundle: nil)
        imContact = nil
        imText = nil
        isNew = true
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        navigationItem.setRightBarButtonItem(UIBarButtonItem(title: "Send", style: .Plain, target: self, action: #selector(ImMessageViewController.onButtonSend(_:))), animated: false)
    }
    
    override func viewWillAppear(animated: Bool) {
        super.viewWillAppear(animated)
        if imContact?.characters.count > 0 && imText?.characters.count > 0 {
            textView.text = imText
            contactAdressBar.text = imContact
        } else {
            
        }
    }
    
    @IBAction func onButtonBack(sender: UIButton) {
        if contactAdressBar.isFirstResponder() {
            contactAdressBar.resignFirstResponder()
        }
        if textView.isFirstResponder() {
            textView.resignFirstResponder()
        }
    }
    
    @IBAction func onButtonSend(sender: UIBarButtonItem) {
        if isNew {
            if contactAdressBar.text!.isEmpty {
                contactAdressBar.becomeFirstResponder()
                return
            }
            if textView.text!.isEmpty {
                textView.becomeFirstResponder()
                return
            }
            
            let appDelegate = AppDelegate.sharedInstance()
            let phone = appDelegate.phone
            
            if phone!.sendTextMessage(contactAdressBar.text, withBody: textView.text) {
                let message = ImMessage()
                message.text = textView.text
                message.from = phone?.config().regUser
                message.to = contactAdressBar.text
                message.isRead = true
                
                appDelegate.addMessage(message)
                appDelegate.storeIms()
            }
            navigationController?.popViewControllerAnimated(true)
        } else {
            isNew = true
            sender.title = "Send"
            textView.text = ""
        }
    }
    
    func textViewDidBeginEditing(textView: UITextView) {
        if !isNew {
            self.textView.text = ""
        }
        isNew = true
        textView.frame = CGRectMake(20, 102, 280, 130)
    }
    
    func textViewDidEndEditing(textView: UITextView) {
        textView.frame = CGRectMake(20, 102, 280, 130)
    }
    
    func textFieldDidBeginEditing(textField: UITextField) {
        if !isNew {
            self.textView.text = ""
        }
        isNew = true
    }
    
    func textFieldShouldReturn(textField: UITextField) -> Bool {
        return textField.resignFirstResponder()
    }
    
    deinit {
        
    }
}
